04 SFC Recipe List/README.md

```diff
@@ -0,0 +1,693 @@
+      'lemon juice',
+      'salt',
+      'pepper'
+    ],
+  },
+  {
+    id: 3,
+    name: 'Spaghetti with tomato sauce',
+    description: `
+     1. Brown beef over medium heat. Drain off fat.
+     2. In a large pot, combine beef, salt, oregano, pepper, garlic powder, onion flakes, diced tomatoes, tomato sauce, and mushrooms. Simmer at a low heat setting for 2 hours, stirring occasionally
+     3. Cook pasta according to package directions. Drain. Serve sauce over spaghetti.
+    `,
+    ingredients: [
+      'spaghetti',
+      '1 beef',
+      '1/4 teaspoon garlic powder',
+      '1 onion',
+      '3 tomatoes',
+      '4 mushrooms',
+      'oregano',
+      'salt',
+    ],
+  },
+];
+
+```
+
+### ./src/rest-api/api/recipe/recipe.ts
+
+```javascript
+import { Recipe } from '../../model';
+import { mockRecipes } from './mockData';
+
+export const fetchRecipes = (): Promise<Recipe[]> => {
+  return Promise.resolve(mockRecipes);
+};
+
+```
+
+### ./src/rest-api/api/recipe/index.ts
+
+```javascript
+export * from './recipe';
+
+```
+
+- Create `recipe` viewModel:
+
+### ./src/pages/recipe/list/viewModel.ts
+
+```javascript
+export interface Recipe {
+  id: number;
+  name: string;
+  description: string;
+  ingredients: string[];
+};
+
+```
+
+- Create `mapper` to map from model to viewModel:
+
+### ./src/pages/recipe/list/mappers.ts
+
+```javascript
+import * as model from '../../../rest-api/model';
+import * as vm from './viewModel';
+
+export const mapRecipeListModelToVm = (recipes: model.Recipe[]): vm.Recipe[] => (
+  recipes.map(mapRecipeModelToVm)
+);
+
+const mapRecipeModelToVm = (recipe: model.Recipe): vm.Recipe => ({
+  ...recipe,
+});
+
+```
+
+- Create `recipe list` page container:
+
+### ./src/pages/recipe/list/PageContainer.vue
+
+```javascript
+<template>
+  <recipe-list-page
+    :recipes="recipes"
+  />
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { Recipe } from './viewModel';
+import { mapRecipeListModelToVm } from './mappers';
+import { fetchRecipes } from '../../../rest-api/api/recipe';
+import RecipeListPage from './Page.vue';
+
+export default Vue.extend({
+  name: 'RecipeListPageContainer',
+  components: {
+    RecipeListPage,
+  },
+  data() {
+    return {
+      recipes: [] as Recipe[],
+    };
+  },
+  created() {
+    fetchRecipes()
+      .then(recipes => {
+        this.recipes = mapRecipeListModelToVm(recipes);
+      })
+      .catch(error => console.log(error));
+  },
+});
+</script>
+
+```
+
+- Create `header`:
+
+### ./src/pages/recipe/list/components/Header.vue
+
+```javascript
+<template>
+  <thead>
+    <th>
+      Name
+    </th>
+    <th>
+      Description
+    </th>
+    <th>
+    </th>
+  </thead>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+export default Vue.extend({
+  name: 'HeaderComponent',
+});
+</script>
+
+```
+
+- To work with CSS Modules, we'll install `webpack-env` typings for use `require`:
+
+```
+npm install @types/webpack-env --save-dev
```

Comment:
If we are using `.vue` files with `<style module>` and `<template> ... :class="$style.name"` maybe we don't need @types/webpack-env.

What do you think?

Reply:
This is from a previous configuration. I deleted it.